src/Lykke.Sdk/LykkeServiceCollectionContainerBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Reflection;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Sdk.Settings;
+using Lykke.SettingsReader;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Lykke.Sdk
+{
+    [PublicAPI]
+    public static class LykkeServiceCollectionContainerBuilderExtensions
+    {
+        public static IServiceProvider BuildServiceProvider<TAppSettings>(this IServiceCollection services,
+            Func<IReloadingManager<TAppSettings>, IReloadingManager<string>> logsConnectionStringFactory)
+            where TAppSettings : BaseAppSettings
+        {
+            var configurationRoot = new ConfigurationBuilder().AddEnvironmentVariables().Build();
+            var settings = configurationRoot.LoadSettings<TAppSettings>();
+
+            var builder = new ContainerBuilder();
+            
+            builder.RegisterModule(new SdkModule<TAppSettings>(settings, logsConnectionStringFactory));
+            builder.RegisterAssemblyModules(Assembly.GetCallingAssembly());
+            builder.Populate(services);
+
+            var container = builder.Build();
+
+            var appLifetime = container.Resolve<IApplicationLifetime>();
+            var log = container.Resolve<ILog>();
+            
+            appLifetime.ApplicationStopped.Register(() =>
```

Comment:
Why there are two ApplicationStopped handlers? Here and here https://github.com/LykkeCity/Lykke.Sdk/pull/1/files#diff-10ea757d90c56c7318d696262ecf67ecR64

Reply:
Watch closely, ApplicationStopped + ApplicationStopping

Reply:
Yep, agree.